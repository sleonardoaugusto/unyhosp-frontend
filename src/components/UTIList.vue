<template>
  <div class="col-md-12 d-flex justify-content-center flex-wrap">
    <div
      v-for="(u, i) in utis"
      :key="i"
      class="card col-md-5"
    >
      <p class="card__name">{{ u.name }}</p>
    </div>
  </div>
</template>

<script>
import utis from '@/mocks/utis'

export default {
  name: 'UTIList',
  data: () => ({
    utis: []
  }),
  created() {
    this.utis = utis.data
  }
}
</script>

<style scoped>

</style>