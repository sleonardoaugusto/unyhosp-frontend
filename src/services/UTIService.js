import apiClient from './config'

export default {
  get() {
    return apiClient.get('/utis/')
  },
  post(data) {
    return apiClient.post('/utis/', data)
  }
}