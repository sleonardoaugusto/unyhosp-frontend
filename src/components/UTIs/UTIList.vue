<template>
  <div class="col-md-12 d-flex justify-content-center flex-wrap">
    <router-link
      v-for="(u, i) in utis"
      :key="i"
      class="card col-md-3"
      :to="{name: 'beds', params: {utiId: u.id}}"
    >
      <div class="row mb-5">
        <p class="card__name">{{ u.name }}</p>
      </div>
    </router-link>
  </div>
</template>

<script>
import UTIService from '@/services/UTIService'

export default {
  name: 'UTIList',
  data: () => ({
    utis: []
  }),
  created() {
    const id = this.$route.params.hospitalId
    UTIService.getByHospitalId(id)
      .then((res) => {
        this.utis = res.data
      })
  }
}
</script>

<style scoped>

</style>