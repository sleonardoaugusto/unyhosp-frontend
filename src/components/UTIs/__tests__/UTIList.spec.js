import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import UTIList from '@/components/UTIs/UTIList'
import utis from '@/components/UTIs/__tests__/__mocks__/utis'
import UTIService from '@/services/UTIService'
import flushPromises from 'flush-promises'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

describe('HospitalList.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(UTIList, {
      localVue,
      router
    })
  })

  it('component is rendering properly', () => {
    wrapper.setData({ utis: utis.data })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('UTIService get should be called on created lifecycle hook', async () => {
    UTIService.get = jest.fn(() => Promise.resolve(utis))
    wrapper = shallowMount(UTIList, { localVue, router })
    await flushPromises()
    expect(wrapper.vm.utis).toBe(utis.data)
  })
})
