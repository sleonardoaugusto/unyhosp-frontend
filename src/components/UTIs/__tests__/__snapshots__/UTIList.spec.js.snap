// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HospitalList.vue component is rendering properly 1`] = `
<div class="col-md-12 d-flex justify-content-center flex-wrap">
  <router-link-stub to="[object Object]" tag="a" event="click" class="card col-md-3">
    <div class="row mb-5">
      <p class="card__name">UTI A - São Joaquim</p>
    </div>
  </router-link-stub>
  <router-link-stub to="[object Object]" tag="a" event="click" class="card col-md-3">
    <div class="row mb-5">
      <p class="card__name">UTI B - São Joaquim</p>
    </div>
  </router-link-stub>
  <router-link-stub to="[object Object]" tag="a" event="click" class="card col-md-3">
    <div class="row mb-5">
      <p class="card__name">UTI C - São Joaquim</p>
    </div>
  </router-link-stub>
</div>
`;
